### ISSUE TITLE
Aggregating function outputs into a `dict`?

### ISSUE THREAD

**Issue by egorchakov** [pipefunc/pipefunc#456]
Hi @basnijholt, thanks for your work on `pipefunc`! 

We've recently started using it in [`rbyte`](https://github.com/yaak-ai/rbyte). Our use case essentially boils down to executing DAGs of functions taking/returning [`polars.DataFrame`](https://docs.pola.rs/api/python/stable/reference/dataframe/index.html)'s or `PyTree`s thereof as part of dataset preprocessing.

One snag I stumbled upon is aggregating a bunch of function outputs into a `dict`. As a first pass I'm resorting to something like [this](https://github.com/yaak-ai/rbyte/blob/main/src/rbyte/utils/_pipefunc.py#L7), [used](https://github.com/yaak-ai/rbyte/blob/main/config/_templates/dataset/yaak.yaml#L62) with [`hydra`](https://hydra.cc/):

```yaml
_target_: pipefunc.PipeFunc
func:
  _target_: hydra.utils.get_method
  path: rbyte.utils.make_dict
bound:
  k0: meta
  k1: mcap
renames:
  v0: meta.data
  v1: mcap.data
output_name: data
```

Is/could there be a better/cleaner way of achieving this with `pipefunc`? Perhaps via some kind of `*args`/`**kwargs` support?

**Comment by basnijholt** [comment]
I am super excited about you using `pipefunc` as a dependency in `rbyte`! 🤩

I'm just boarding a plane but will have a detailed look and answer your question ASAP.

Perhaps I can add some native helper code to make your use case more naturally supported.

**Comment by basnijholt** [comment]
@egorchakov I opened this PR (while on the ✈️ ) https://github.com/pipefunc/pipefunc/pull/457. Please let me know whether you think this will solve your usecase!

I added an example in the FAQ about how to use it.

**Comment by basnijholt** [comment]
The satellite internet I am on is too slow to clone `yaak-ai/rbyte` and install its dependencies, so I cannot try it out in your code base.

I am confident you'll get it to work though, if not we'll try something else 😉

**Comment by egorchakov** [comment]
Thank you, this looks like exactly what I needed! Will try it out once I'm back from vacation.

**Comment by basnijholt** [comment]
I merged #457 and released a new version. 

The [FAQ](https://pipefunc.readthedocs.io/en/latest/faq/#how-to-collect-results-as-a-step-in-my-pipeline) should explain it well!

Let me know if you have any more suggestions or experience pain points. I'd be happy to solve them!
